crates/typst-html/src/document.rs

```diff
@@ -126,60 +126,98 @@ fn introspect_html(
 ) -> Introspector {
     fn discover(
         builder: &mut IntrospectorBuilder,
-        sink: &mut Vec<(Content, Position)>,
+        sink: &mut Vec<(Content, DocumentPosition)>,
         link_targets: &mut FxHashSet<Location>,
         nodes: &[HtmlNode],
+        current_position: &mut EcoVec<usize>,
     ) {
+        let mut index = 0;
         for node in nodes {
             match node {
                 HtmlNode::Tag(tag) => {
+                    current_position.push(index);
                     builder.discover_in_tag(
                         sink,
                         tag,
-                        Position { page: NonZeroUsize::ONE, point: Point::zero() },
+                        HtmlPosition { element: current_position.clone(), inner: None },
```

Comment:
storing a full vector here for every tag might become expensive. we don't necessarily have to do something about this now, but worth discussing. perhaps also an indication that we should turn the html position into a type with private fields + an iterator method, so that we are more free to adjust the representation later (for example, into a tree structure with shared parts).